### PR TITLE
[sql mode] Added "text/x-sqlite" mime type

### PR DESCRIPTION
**Pull request by crazy4chrissi**
Added the "text/x-sqlite" mime type for sql-mode. Neither existing mime-types mysql nor x-sql fit well for SQLite, e.g. important keywords like PRAGMA are missing. Therefore, I added a new mime type based on the SQLite language definition available online. I added references to all sources in comments. Feel free to remove them if you think this is too much.
SQLite says identifiers like columns should be quoted in double quotes preferably, just like in standard SQL, not backticks. Therefore, I made the necessary changes to sql-hint.js, so that a mime type declaration can define the column identifier that the language uses. I guess other SQL dialects will need this as well.

### CODE REVIEW

**Comment by crazy4chrissi** [comment]
Ok sorry, I will fix these.

File: addon/hint/sql-hint.js [hunk 1]
```diff
@@ -28,6 +29,15 @@
     return CodeMirror.resolveMode(mode).keywords;
   }
 
+  function getIdentifierQuote(editor) {
+    var mode = editor.doc.modeOption;
```
Comment (marijnh) [inline addon/hint/sql-hint.js]: It's easier to do `editor.getMode()` to get the mode (and then check `mode.name == "sql"`)

**Comment by marijnh** [comment]
Could you address my comment, add a minimal test for handling double-quoted identifiers to `test/sql-hint-tests.js`, and add this new dialect to `mode/meta.js`?

**Comment by crazy4chrissi** [comment]
Thanks for your review. I will try to do a test case and add it to meta.js. Regarding getting the mode: I did it the same way as  function getKeywords(editor) in the same file does. So then this should be adjusted there as well I think?
```
  function getKeywords(editor) {
    var mode = editor.doc.modeOption;
    if (mode === "sql") mode = "text/x-sql";
    return CodeMirror.resolveMode(mode).keywords;
  }
```

**Comment by marijnh** [comment]
Hm, on closer look, no, it makes sense the way it is now -- you're calling `resolveMode`, not `getMode`, because you need the original mode spec, not the mode object, to be able to get at its configuration. So the code is fine as it is (though I'd simplify the end of the function to ``CodeMirror.resolveMode(mode).identifierQuote) || "`"``

**Comment by crazy4chrissi** [comment]
Thank you. I will do like this. I am currently looking at the `sql-hint-tests.js` and if I am not mistaken it has some problems. This test does not fail:
```
  test("keywords2", {
    value: "SEA",
    cursor: Pos(0, 3),
    list: ["SELECT"],
    from: Pos(0, 0),
    to: Pos(0, 3)
  });
```
The only change I did from the original `keywords` test is to change SEL to SEA. Shouldn't this complain that "SELECT" is not in the list of autocompletion as required by the test `keywords2`?

**Comment by marijnh** [comment]
You're right, those tests weren't failing when the expected result had more elements than the produced result. Patch b359a021 fixes this.

**Comment by crazy4chrissi** [comment]
Ok all done, anything else?

**Comment by marijnh** [comment]
Hi, the tests have syntax errors (missing commas) and one of them fails.

**Comment by crazy4chrissi** [comment]
I am sorry. I cannot reproduce this. I just did a fresh clone and only exchanged the `test/sql-hint-test.js`.
- ran bin/lint without problems
- continuous-integration does not report any problem
- `npm test` says "All passed", but this was also the case when tests should fail!?
- test/index.html reports that 1 test called `core_lineWidgets`fails, but this is the same with a fresh clone and seems to be unrelated to my tests

Can you please explain where you found syntax errors and which test is failing for you?

File: test/sql-hint-test.js [hunk 2]
```diff
@@ -99,6 +109,16 @@
     to: Pos(0, 16)
   });
 
+  test("doublequotedcolumn", {
+    value: "SELECT users.\"na",
+    cursor: Pos(0, 16),
+    tables: simpleTables,
+    list: ["\"users\".\"name\""],
+    from: Pos(0, 7),
+    to: Pos(0, 16)
```
Comment (marijnh) [inline test/sql-hint-test.js]: Missing comma here

File: test/sql-hint-test.js [hunk 3]
```diff
@@ -118,6 +138,16 @@
     to: Pos(0, 11)
   });
 
+  test("schemadoublequoted", {
+    value: "SELECT \"sch",
+    cursor: Pos(0, 11),
+    tables: schemaTables,
+    list: ["\"schema\".\"users\"", "\"schema\".\"countries\""],
+    from: Pos(0, 7),
+    to: Pos(0, 11)
```
Comment (marijnh) [inline test/sql-hint-test.js]: Another one here, and the test fails (no results)

**Comment by marijnh** [comment]
I'm not sure what you're doing, but since the test file is syntactically invalid, you can't have been running it (probably you missed an error on the console saying that the file failed to load, and thus the tests were skipped)

**Comment by crazy4chrissi** [comment]
Okay you are right, but I still don't get it why nothing on my side nor the continuous-integration check complained about the syntax error.  Attached you will find exactly what I did:
[test.txt](https://github.com/codemirror/CodeMirror/files/810578/test.txt)
I just saw that Firefox complains about the syntax error in the console when running the test/index.html. So next time I should look there as well....
Anyway. I will soon commit a fix of the tests.

**Comment by marijnh** [comment]
The linter apparently doesn't check the test scripts, and I guess the phantomjs process also failed to evaluate the broken file and thus didn't define or run the tests.

**Comment by marijnh** [comment]
Thanks, looks great now. Merged as 65c70cf5d18 and c4387d6073b15

**Comment by crazy4chrissi** [comment]
Great thank you for all your great work. I am going to integrate CodeMirror into phpLiteAdmin, a web-based GUI to manage SQLite databases. Therefore, I need good highlighting of SQLite in CodeMirror.

**Pull request closed without merging**